web_server.go

```diff
@@ -58,15 +61,21 @@ func newWebServer(
 		return nil, fmt.Errorf("failed reading default index.html page: %v", err)
 	}
 
-	fs := handleStatic([]byte(mungeIndexPage(baseURL, string(indexPage))), http.StripPrefix(baseURL, http.FileServer(http.Dir(webDir))), header)
+	fs := handleStatic([]byte(mungeIndexPage(baseURL, string(indexPage), previewImage, previewTitle, previewDescription)), http.StripPrefix(baseURL, http.FileServer(http.Dir(webDir))), header)
 	if auth != nil {
 		fs = auth.wrap(fs)
 	}
 	return &WebServer{site, broker, fs, keychain, maxRequestSize, baseURL}, nil
 }
 
-func mungeIndexPage(baseURL, html string) string {
+func mungeIndexPage(baseURL, html string, previewImage string, previewTitle string, previewDescription string) string {
```

Comment:
nit: Should be enough.

```suggestion
func mungeIndexPage(baseURL, html, previewImage, previewTitle, previewDescription string) string {
```